snuba/datasets/transactions_processor.py

```diff
@@ -0,0 +1,115 @@
+from datetime import datetime
+
+import ipaddress
+import uuid
+
+from snuba.processor import (
+    _as_dict_safe,
+    MessageProcessor,
+    _ensure_valid_date,
+    _unicodify
+)
+from snuba.datasets.events_processor import (
+    enforce_retention,
+    extract_base,
+    extract_extra_contexts,
+    extract_extra_tags,
+    extract_user,
+)
+
+
+class TransactionsMessageProcessor(MessageProcessor):
+    PROMOTED_TAGS = [
+        "environment",
+        "sentry:release",
+        "sentry:user",
+        "sentry:dist",
+    ]
+
+    def __extract_timestamp(self, field):
+        timestamp = _ensure_valid_date(datetime.fromtimestamp(field))
+        milliseconds = int(timestamp.microsecond / 1000)
+        return (timestamp, milliseconds)
+
+    def process_message(self, message, metadata=None):
+        action_type = self.INSERT
+        processed = {'deleted': 0}
+        if not isinstance(message, (list, tuple)) and len(message) >= 2:
```

Comment:
I don't understand this conditional. Shouldn't this be an `or` and not an `and`? Otherwise this would let things like strings with 2 or more characters through?